overlord/snapstate/snapstate.go

```diff
@@ -354,6 +356,25 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		ts.MarkEdge(checkAsserts, DownloadAndChecksDoneEdge)
 	}
 
+	if preseedMode() {
```

Comment:
we should always do this, or have a flags value to control this, also this code probably needs a sanity unit test in this package

Reply:
Right, thanks for spotting, it was already always done unconditionally below but probably due to a wrong merge conflict resolution I messed it up and ended up doing this twice. Simplified now. The test is already there in TestInstallTaskEdgesForPreseeding.